CHANGELOG.d/breaking_fail-on-cjs-detected.md

```diff
@@ -0,0 +1,7 @@
+* Fail if CJS is detected in FFI
+
+  Previously, one could write CommonJS in FFI declarations. With the move to ES modules, this leads to a failing compile now.
+
+* Do not emit an error if language-javascript parsing fails
+
+  Previously, an error was emitted when language-javascript parsing failed. With this change, no error or warning is emitted if the js FFI parsing fails.
```

Comment:
Two comments here. First (and just for future PRs), if this is supposed to be two separate entries, then it should be two separate files.

Second, I think this is actually one entry. Perhaps this rendering summarizes things better? I'd like to hear other core team members' feedback on this point.
```markdown
* If FFI parsing succeeds & CommonJS is detected, fail; otherwise, do not error or warn

  Previously, the compiler would emit an error if it failed to parse the FFI JavaScript file.
  Since the underlying JavaScript parser (i.e. `language-javascript`) fails to parse even
  valid JavaScript files, we cannot consider every failed parse to mean invalid JS files.
  Fixing the parser would require a lot of effort, so we are planning to remove it instead
  in `v0.16.x`.
  
  If the parse succeeds and a CommonJS module is detected, a compiler error is now emitted.
  If the parse fails, we no longer emit a compiler error. While we could emit a warning,
  such a warning will quickly become annoying for FFI files that trigger the buggy paths
  of `language-javascript`. Moreover, we presume that all will be migrating their code to
  ES modules now that CommonJS is being deprecated in the larger JavaScript ecosystem.
```

> CommonJS is being deprecated in the larger JavaScript ecosystem

Regarding this point in the above entry, @sigma-andex could you link to the post that mentioned this? I can't recall where it is.

Reply:
I think you meant [this discussion](https://github.com/sindresorhus/meta/discussions/15). And there are many libraries that follow this, like e.g. [`node-fetch`](https://github.com/node-fetch/node-fetch#commonjs).

Reply:
Should I add this to the changelog?

Reply:
I don't think that was the post I remember you mentioning elsewhere. But since I can't find that post, yeah, let's link to that discussion.